### PR TITLE
SDFSOF-67 - Updated Custody Server and Observer to use new RPC endpoint

### PR DESCRIPTION
**Pull request by MazurakIhor**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Updated Custody Server and Observer to use new RPC endpoint

As for now, target branch of this PR is set to fireblocks_integration but it will be changed to softserve_develop as soon as https://github.com/stellar/java-stellar-anchor-sdk/pull/1002 is approved and merged.

Changed validation of input payments (assets, amounts) to not set transaction to error status but just log message since most likely business will need to make refund of received payments if something went wrong. Business can still change state to error using notify_transaction_error action if needed.

### Why

All transaction updates should be done using new RPC actions endpoint. In scope of this task PATCH /transactions requests sent by Custody Server and Observer were replaced with corresponding RPC calls

### CODE REVIEW

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 1]
```diff
@@ -711,3 +711,15 @@ data:
   #
   flyway_baseline_on_migrate: false
 
+################################
+## RPC Configuration
+################################
+rpc:
```
Comment (rkharevych) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:717]: How about `rpc.actions.custom_mesages`?

File: platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java [hunk 2]
```diff
@@ -89,9 +119,6 @@ protected void validatePayment(JdbcCustodyTransaction txn, CustodyPayment paymen
           payment.getAssetType(),
           payment.getId(),
           payment.getExternalTxId());
-      payment.setStatus(CustodyPaymentStatus.ERROR);
```
Comment (rkharevych) [inline platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java:92]: I think, that we can go without this validation right now. We will update it, if it is needed. Check whether we can remove `message` and `status` from `CustodyPayment`

**Review by rkharevych — CHANGES_REQUESTED** [review]

**Review by rkharevych — DISMISSED** [review]

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 1 above)
Comment (MazurakIhor) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:717]: changed

File: platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java [hunk 3]
```diff
@@ -75,7 +73,39 @@ protected void updateTransaction(
     txn.setStatus(getCustodyTransactionStatus(payment.getStatus()).toString());
     custodyTransactionRepo.save(txn);
 
-    patchTransaction(txn, payment, newSepTxnStatus);
+    if (SepTransactionStatus.ERROR.equals(newSepTxnStatus)) {
```
[the remark this answers is not in the document]
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java]: Changed

File: platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java (hunk 2 above)
Comment (MazurakIhor) [reply platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java:92]: Status is still used.

**Review by Ifropc — APPROVED** [review]

File: platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java [hunk 4]
```diff
@@ -75,7 +72,27 @@ protected void updateTransaction(
...
+    } else {
+      switch (Kind.from(txn.getKind())) {
+        case DEPOSIT:
+          platformApiClient.notifyOnchainFundsSent(
+              txn.getSepTxId(),
+              payment.getTransactionHash(),
+              rpcConfig.getActions().getCustomMessages().getOutgoingPaymentSent());
+          break;
+        case WITHDRAWAL:
+        case RECEIVE:
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/custody/CustodyPaymentHandler.java:87]: Won't think throw an exception on platform server as SEP-31 not supported yet?

**Pull request merged**